x/staking/client/cli/tx.go

```diff
@@ -79,8 +79,12 @@ func NewCreateValidatorCmd(clientCtx client.Context) *cobra.Command {
 	cmd.Flags().String(FlagNodeID, "", "The node's ID")
 
 	_ = cmd.MarkFlagRequired(flags.FlagFrom)
-	_ = cmd.MarkFlagRequired(FlagAmount)
-	_ = cmd.MarkFlagRequired(FlagPubKey)
+
+	// TODO: Investigate why the the following two required flag, "FlagAmount" and "FlagPubKey", are force to the genTx cmd in x/genutil/client/cli/gentx.go.
```

Comment:
I don't understand this exactly. What is the issue?

Reply:
If I  uncomment these two line of code:
```
_ = cmd.MarkFlagRequired(FlagAmount)
_ = cmd.MarkFlagRequired(FlagPubKey)
```
then some of the integration tests will fail like this one:

```
TestCLISimdCollectGentxs: gobash.go:60: Running /home/dau/workspace/go/src/github.com/cosmos/cosmos-sdk/build/simd gentx --name=foo --home=/tmp/sdk_integration_TestCLISimdCollectGentxs_726251297/.simd --home-client=/tmp/sdk_integration_TestCLISimdCollectGentxs_726251297/.simcli --keyring-backend=test --output-document=/tmp/167445516/gentx.json
    TestCLISimdCollectGentxs: executors.go:41: Stderr: ERROR: required flag(s) "amount", "pubkey" not set
        
    TestCLISimdCollectGentxs: executors.go:13: 
        	Error Trace:	executors.go:13
        	            				helpers.go:78
        	            				simd_test.go:53
        	Error:      	Should be true
        	Test:       	TestCLISimdCollectGentxs
--- FAIL: TestCLISimdCollectGentxs (2.34s)
```

Reply:
Should we not investigate and fix those issues?

Reply:
The issue got resolved automatically after implementing the feedback. :)

Reply:
No, my bad. The the integration tests are still failing because of this issue. It seems that the command `gentx` share the same flags from the staking module:

![image](https://user-images.githubusercontent.com/30392990/86065593-45e59a00-ba70-11ea-978b-35e969656701.png)

the staking module fullfill the `StakingMsgBuildingHelpers` interface

![image](https://user-images.githubusercontent.com/30392990/86065734-a96fc780-ba70-11ea-8e4d-b319c306dfe9.png)

and in the end it get all these flags:

![image](https://user-images.githubusercontent.com/30392990/86065888-053a5080-ba71-11ea-8958-4a73105c5b30.png)

![image](https://user-images.githubusercontent.com/30392990/86066038-58ac9e80-ba71-11ea-8acc-e4e2e0eb4ba1.png)

the same flags are share in the `NewCreateValidatorCmd` from the staking module

![image](https://user-images.githubusercontent.com/30392990/86066236-e8524d00-ba71-11ea-895e-5b77c036c02f.png)

but in  `NewCreateValidatorCmd` we mark the flag `FlagAmount`, `FlagPubKey` as required and this annotations is automatically applied to the command `gentx` .

**Is this behaviour supposed to be correct? If yes we can simply fix the test cases. If not, I need the help of someone to fix this.**

In the current master branch the only required flag for `gentx` is `FlagName`.

Reply:
Perhaps I know why the reason of this issue:
The flags `FsPk` and `FsAmount` are global variable and both `gentx` and `NewCreateValidatorCmd``are accessing it but the latest one has marked then as required and this affect also `gentx`. 

I don't think sharing global variables is such a great thing. We should avoid it as much as possible. Lete see how we can fix this...

Reply:
Agree on the global vars, we should get rid of them. :D